File: sr-link-git-module-framework/playwright-java-bdd-testing/src/main/java/com/srlink/playwright/pages/DashboardPage.java
package com.srlink.playwright.pages;

import com.microsoft.playwright.Page;

public class DashboardPage {

  private final Page page;
  private final String userMenuSelector = ".oxd-userdropdown-tab";
  private final String logoutSelector = "a:has-text('Logout')";

  public DashboardPage(Page page) {
    this.page = page;
  }

  public boolean isLoaded() {
    return page.locator(userMenuSelector).isVisible();
  }

  public LoginPage logout() {
    page.click(userMenuSelector);
    page.click(logoutSelector);
    return new LoginPage(page);
  }
}

File: orange-hrm-framework/src/pages/DashboardPage.ts
import { Page, Locator } from '@playwright/test';

export class DashboardPage {
  readonly page: Page;
  readonly userDropdown: Locator;
  readonly logoutLink: Locator;
  readonly dashboardHeader: Locator;

  constructor(page: Page) {
    this.page = page;
    this.userDropdown = page.locator('.oxd-userdropdown-tab');
    this.logoutLink = page.locator('a:has-text("Logout")');
    this.dashboardHeader = page.locator('.oxd-topbar-header-breadcrumb');
  }

  async isLoaded(): Promise<boolean> {
    try {
      await this.userDropdown.waitFor({ timeout: 10000 });
      return true;
    } catch {
      return false;
    }
  }

  async logout() {
    await this.userDropdown.click();
    await this.logoutLink.click();
  }

  async getHeaderText(): Promise<string | null> {
    try {
      return await this.dashboardHeader.textContent();
    } catch {
      return null;
    }
  }
}
